knox.go

```diff
@@ -52,15 +61,62 @@ func (c *knoxClient) RequestCredential(cred_type string) (credential_adapter.Ver
 	return cred, nil
 }
 
-func (c *knoxClient) PresentCredential(cred credential_adapter.VerifiableCredential) error {
+func (c *knoxClient) PresentCredential(creds ...model.SerializedDocument) error {
+
+	challenge, err := c.ca.CreatePresentationChallenge()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Challenge: %s, %s\n", challenge.Nonce, challenge.Url)
+	converted_creds := make([]map[string]interface{}, len(creds))
+
+	for i, cred := range creds {
+		var converted_cred map[string]interface{}
+		err = json.Unmarshal(cred, &converted_cred)
+
+		if err != nil {
```

Comment:
NIT: one liner with `if err := json.Unmarshal(cred, &converted_cred);  err != nil{`

Reply:
I'm generally against complex one-liners but this is simple enough that making it a one-liner doesn't make it harder to read and flow